test/jdk/java/awt/print/PrinterJob/ValidatePage/ValidatePage.java

```diff
@@ -21,407 +21,293 @@
+        pp.add(pgih = new Label());
+        pp.add(pgrm = new Label());
+        pp.add(pgbm = new Label());
+
+        add(pp);
+
+        Panel epp = new Panel();
+        epp.setLayout(new GridLayout(6, 2, 0, 0));
+
+        epp.add(new Label("Page width:"));
+        epp.add(tpw = new TextField());
+        epp.add(new Label("Page height:"));
+        epp.add(tph = new TextField());
+        epp.add(new Label("Left Margin:"));
+        epp.add(tpglm = new TextField());
+        epp.add(new Label("Top margin:"));
+        epp.add(tpgtm = new TextField());
+        epp.add(new Label("Imageable Wid:"));
+        epp.add(tpgiw = new TextField());
+        epp.add(new Label("Imageable Hgt:"));
+        epp.add(tpgih = new TextField());
+
+        add(epp);
+        displayPageFormatAttributes();
+
+        Panel panel = new Panel();
+        Button defButton = new Button("Default Page");
+        defButton.addActionListener(e -> {
+            myPageFormat = myPrinterJob.defaultPage();
+            displayPageFormatAttributes();
+        });
+
+        Button pageButton = new Button("Page Setup..");
+        pageButton.addActionListener(e -> {
+            myPageFormat = myPrinterJob.pageDialog(myPageFormat);
+            displayPageFormatAttributes();
+        });
+        Button printButton = new Button("Print");
+        printButton.addActionListener(e -> {
+            try {
+                myPrinterJob.setPrintable(ValidatePage.this, myPageFormat);
+                myPrinterJob.print();
+            } catch (PrinterException pe) {
+                PassFailJFrame.forceFail( "Test Failed");
+                pe.printStackTrace();
+            }
+        });
+
+        Button chooseButton = new Button("Printer..");
+        chooseButton.addActionListener(e -> myPrinterJob.printDialog());
+
+        Button validateButton = new Button("Validate Page");
+        validateButton.addActionListener(e -> {
+            myPageFormat = myPrinterJob.validatePage(myPageFormat);
+            displayPageFormatAttributes();
+        });
+        Button setButton = new Button("Set Paper");
+        setButton.addActionListener(e -> {
+            try {
+                Paper p1 = new Paper();
+                double pwid = Double.parseDouble(tpw.getText());
+                double phgt = Double.parseDouble(tph.getText());
+                double pimx = Double.parseDouble(tpglm.getText());
+                double pimy = Double.parseDouble(tpgtm.getText());
+                double pimwid = Double.parseDouble(tpgiw.getText());
+                double pimhgt = Double.parseDouble(tpgih.getText());
+                p1.setSize(pwid, phgt);
+                p1.setImageableArea(pimx, pimy, pimwid, pimhgt);
+                myPageFormat.setPaper(p1);
+                displayPageFormatAttributes();
+            } catch (NumberFormatException nfe) {
+                PassFailJFrame.forceFail( "Test Failed");
+                nfe.printStackTrace();
+            }
```

Comment:
This should rather display a warning to the tester so that they're able to correct it and continue with the test.